src/lib/steps/package.ts

```diff
@@ -118,57 +118,78 @@ async function findSecondaryPackagePaths(rootPackage: NgPackageData): Promise<st
 }
 
 /**
- * Reads an Angular package definition first from the passed in file path,
- * then from the default ng-package.json file,
- * then from package.json, and merges the json into one config object.
+ * Reads an Angular package, i.e. `NgPackageConfig`, from the passed in path.
+ *
+ * The `projectPath` can either be a file path:
+ *  - a `package.json` with custom `ngPackage` property
+ *  - an `ng-package.json` file representing `NgPackageConfig`
+ *  - an `ng-package.js` exporting `NgPackageConfig`
  *
- * @param filePath path pointing to `ng-package.json` file
+ * If `projectPath` is a path to a directory, configuration will be read in sequential order from:
+ *  - `package.json`,
+ *  - `ng-package.json`,
+ *  - `ng-package.js`.
+ *
+ * @param projectPath Path pointing to `package.json` or `ng-package.json` or `ng-package.js` file
  */
-async function readRootPackage(filePath: string): Promise<NgPackageData> {
+async function readRootPackage(projectPath: string): Promise<NgPackageData> {
 
   const cwd: string = process.cwd();
-  if (!path.isAbsolute(filePath)) {
-    filePath = path.resolve(cwd, filePath);
+  if (!path.isAbsolute(projectPath)) {
+    projectPath = path.resolve(cwd, projectPath);
   }
 
-  const baseDirectory = path.dirname(filePath);
-
-  // read custom ng-package config file
-  let promiseChain: Promise<NgPackageConfig> = readNgPackageFile(filePath)
-    .then((ngPkg: NgPackageConfig | null) => ngPkg || {});
-
-  const defaultPath = path.join(baseDirectory, 'ng-package.json');
-  if (defaultPath !== filePath) {
-    // read default ng-package config file
-    promiseChain = promiseChain.then(async (ngPkg: NgPackageConfig) => {
-      const otherNgPkg: NgPackageConfig | null = await readNgPackageFile(defaultPath);
-      // merge both ng-package config objects
-      // merge will never return null
-      return merge(ngPkg, otherNgPkg, arrayMergeLogic);
-    });
+  let ngPkg: NgPackageConfig = {};
+  let baseDirectory: string;
+
+  const stats = await lstat(projectPath);
+  if (stats.isDirectory()) {
+    // Read from directory
+    baseDirectory = projectPath;
+    try {
```

Comment:
You could make the read logic into a function to avoid repeating yourself multiple times here.
```ts
let ngPkg: NgPackageConfig = {};
const fileNames = [{
    name: 'package.json',
    resolverFunc: (pkgJson) => pkgJson.ngPackage.
  }, 
  'ng-package.json',
  'ng-package.js'
];
for(const fileResolveArg of fileNames) {
  ngPkg = ReadConfig(fileResolveArg);
}
```